### PR TITLE
Fix wrong orientation when there isn't at least 2 elements on the list

### PR DESCRIPTION
**Pull request by scout719**

### CODE REVIEW

File: src/react-virtualized-list/virtualized-scroll-viewer.tsx [hunk 1]
```diff
@@ -360,9 +360,16 @@ export class VirtualizedScrollViewer extends React.Component<IScrollViewerProper
...
+        let children = itemsContainer.children;
+
         // ignore spacer elements
-        for (let i = 1; i < itemsContainer.children.length - 1; i++) {
-            items.push(itemsContainer.children[i]);
+        let start_idx = children.length > 0 && Spacer.isSpacer(children[0]) ? 1 : 0;
+        let end_idx = children.length > 0 && Spacer.isSpacer(children[children.length - 1]) ? children.length - 2 : children.length - 1;
+        for (let i = start_idx; i < end_idx; i++) {
+            var elem = itemsContainer.children[i];
+            if (!Spacer.isSpacer(elem)) {
```
Comment (joaompneves) [inline src/react-virtualized-list/virtualized-scroll-viewer.tsx]: Why do you have to check very element? Its not supposed to have spacers in between

File: src/react-virtualized-list/virtualized-scroll-viewer.tsx [hunk 2]
```diff
@@ -360,9 +360,16 @@ export class VirtualizedScrollViewer extends React.Component<IScrollViewerProper
      */
     private getListItems(itemsContainer: HTMLElement): Element[] {
         let items: Element[] = [];
+        let children = itemsContainer.children;
+
         // ignore spacer elements
```
Comment (joaompneves) [inline src/react-virtualized-list/virtualized-scroll-viewer.tsx]: if (children.length > 0) {
  ...

File: src/react-virtualized-list/spacer.tsx [hunk 3]
```diff
@@ -26,13 +27,14 @@ export class Spacer extends React.Component<SpacerProps, {}>{
...
+            style.height = dimension + PIXEL_UNITS;
             style.width = FILL_SPACE;
         }
-        
-        
-        // style.backgroundColor = "#f0f";
-        
-        return React.DOM.script({ key: childKey, style: style });
+
+        return React.DOM.script({ key: childKey, style: style, data: SPACER_ATTRIBUTE });
```
Comment (joaompneves) [inline src/react-virtualized-list/spacer.tsx]: instead of an attribute use a ref

**Review by joaompneves — CHANGES_REQUESTED** [review]

File: src/react-virtualized-list/spacer.tsx (hunk 3 above)
Comment (scout719) [reply src/react-virtualized-list/spacer.tsx]: I tried it but I don't know how to work with ReactFragments and so. Care to help me?

File: src/react-virtualized-list/virtualized-scroll-viewer.tsx [hunk 4]
```diff
@@ -360,9 +360,16 @@ export class VirtualizedScrollViewer extends React.Component<IScrollViewerProper
...
     private getListItems(itemsContainer: HTMLElement): Element[] {
         let items: Element[] = [];
-        // ignore spacer elements
-        for (let i = 1; i < itemsContainer.children.length - 1; i++) {
-            items.push(itemsContainer.children[i]);
+        let children = itemsContainer.children;
+
+        if (children.length > 0) {
+            // ignore spacer elements
+            let start_idx = Spacer.isSpacer(children[0]) ? 1 : 0;
```
Comment (oscsx) [inline src/react-virtualized-list/virtualized-scroll-viewer.tsx]: start_idx -> startIdx
end_idx -> endIdx

**Review by oscsx — CHANGES_REQUESTED** [review]
variable casing

File: src/react-virtualized-list/virtualized-scroll-viewer.tsx [hunk 5]
```diff
@@ -364,8 +367,8 @@ export class VirtualizedScrollViewer extends React.Component<IScrollViewerProper
 
         if (children.length > 0) {
             // ignore spacer elements
-            let start_idx = Spacer.isSpacer(children[0]) ? 1 : 0;
-            let end_idx = Spacer.isSpacer(children[children.length - 1]) ? children.length - 2 : children.length - 1;
+            let start_idx = this.isSpacer(children[0]) ? 1 : 0;
```
Comment (joaompneves) [inline src/react-virtualized-list/virtualized-scroll-viewer.tsx]: startIdx
endIdx

**Review by joaompneves — DISMISSED** [review]

**Review by oscsx — DISMISSED** [review]

**Review by oscsx — APPROVED** [review]

**Review by joaompneves — APPROVED** [review]

**Pull request merged**